src/display.jl

```diff
@@ -75,7 +75,22 @@ end
 plot(args...; kwargs...) = SyncPlot(Plot(args...; kwargs...))
 
 ## API methods for SyncPlot
-for f in [:restyle!, :relayout!, :addtraces!, :deletetraces!, :movetraces!,
+function restyle!(sp::SyncPlot, args...; kwargs...)
+    diffs = restyle!(sp.plot, args...; kwargs...)
+    # each trace in the plot returns a diff, so we have to apply them
+    # separately. Maybe at some point we can do some clever merging
+    for (i, diff) in enumerate(diffs)
```

Comment:
This is a good first pass, but I think that we'll eventually want to merge all the diffs on the Julia side so we aren't sending many little diffs over the server to js.

I could be wrong, but I anticipate that performance would be better if we did the merging in Julia